packages/xo-web/src/xo-app/index.js

```diff
@@ -199,7 +199,11 @@ export const ICON_POOL_LICENSE = {
 
         for (const host of hosts) {
           const license = xcpngLicenseByBoundObjectId[host.id]
-          if (license !== undefined && license.expires > Date.now()) {
+          if (
+            license !== undefined &&
+            (license.expires = license.expires ?? Infinity) &&
+            license.expires > Date.now()
+          ) {
```

Comment:
Extract the assignment it in a separate condition for readability, so we know what state we're handling.